plugins/output/git/git.go

```diff
@@ -0,0 +1,49 @@
+package git
+
+import (
+	"os"
+
+	"github.com/go-git/go-git/v5"
+
+	"github.com/parinithshekar/github-migration-cli/common"
+	plugins "github.com/parinithshekar/github-migration-cli/plugins/interfaces"
+)
+
+// Client struct has the output plugin associated with the integration
+type Client struct {
+	output *plugins.Output
+}
+
+// New returns a new git instance to perform git functions
+func New(output *plugins.Output) (*Client) {
+	gitClient := new(Client)
+	gitClient.output = output
+	return gitClient
+}
+
+// SyncRepos clones repositories locally and syncs 
+func (gitClient Client) SyncRepos(repos []common.Repository) {
+	if _, err := os.Stat("syncDirectory"); os.IsNotExist(err) {
+		os.Mkdir("syncDirectory", os.ModeDir)
+	}
+	os.Chdir("syncDirectory")
+	
+	for _, repo := range repos {
+		if _, err := os.Stat(repo.Slug); os.IsNotExist(err) {
+			git.PlainClone(repo.Slug, false, &git.CloneOptions{ URL: repo.Source })
+		}
+		os.Chdir(repo.Slug)
+		gitClient.SyncTags(repo)
+		gitClient.SyncBranches(repo)
+	}
+}
+
+// SyncTags individually syncs the tags from source remote to the target remote
+func (gitClient Client) SyncTags(repo common.Repository) {
+	// Sync Tags individually
```

Comment:
May not be necessary to sync each tag or branch individually.  If you can run the pull command, and then parse the output to understand what individual tags or branches failed or succeeded, and then warn appropriately, then the bulk sync commands could work just fine.

You'd have to be able to be able to handle errors from those bulk commands. For example, if someone force-pushed a branch, this program needs to identify which branch failed and why amongst the bulk command.

Reply:
Yeah makes sense, I'll experiment with what kind of errors the go-git library gives out. Should be able to work this out.